src/utils.cpp

```diff
@@ -778,4 +779,17 @@ std::unordered_set<std::string> get_traceable_funcs()
   return result;
 }
 
+uint64_t parse_exponent(const char *str)
+{
+  char *e_offset;
+  auto base = strtoll(str, &e_offset, 0);
+
+  if (*e_offset != 'e')
+    return base;
+
+  auto exp = strtoll(e_offset + 1, nullptr, 0);
```

Comment:
```suggestion
  auto exp = strtoll(e_offset + 1, nullptr, 10);
```
I think the base should be 10. 1e010 is 100_000_000 for now, but it is 10_000_000_000 in C.

Reply:
yeah might be better. I've updated it